client/client.go

```diff
@@ -214,6 +188,36 @@ func (c *LogClient) UpdateRoot(ctx context.Context) error {
 	return nil
 }
 
+// WaitForInclusion blocks until the requested data has been verified with an inclusion proof.
+func (c *LogClient) WaitForInclusion(ctx context.Context, data []byte) error {
+	leaf := c.logVerifier.buildLeaf(data)
+
+	// Fetch the current Root to improve our chances at a valid inclusion proof.
+	if err := c.UpdateRoot(ctx); err != nil {
+		return err
+	}
+	if c.root.TreeSize == 0 {
+		return fmt.Errorf("Tree is empty, TreeSize = 0")
```

Comment:
Why is this necessarily an error?  What if you used `AddLeaf` to add the very first leaf to a log?